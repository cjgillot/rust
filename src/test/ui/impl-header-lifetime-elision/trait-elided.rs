#![allow(warnings)]

trait MyTrait<'a> {}

impl MyTrait for u32 {
    //~^ ERROR implicit elided lifetime not allowed here
    //~| ERROR missing lifetime specifier [E0106]
}

fn main() {}
