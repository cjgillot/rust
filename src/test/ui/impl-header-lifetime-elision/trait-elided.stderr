error[E0726]: implicit elided lifetime not allowed here
  --> $DIR/trait-elided.rs:5:6
   |
LL | impl MyTrait for u32 {
   |      ^^^^^^^- help: indicate the anonymous lifetime: `<'_>`
   |
   = note: assuming a `'static` lifetime...

error[E0106]: missing lifetime specifier
  --> $DIR/trait-elided.rs:5:6
   |
LL | impl MyTrait for u32 {
   |      ^^^^^^^ expected named lifetime parameter
   |
help: consider introducing a named lifetime parameter
   |
LL | impl<'a> MyTrait<'a> for u32 {
   |     ++++ ~~~~~~~~~~~

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0106`.
